connexion/middleware/main.py

```diff
@@ -0,0 +1,65 @@
+import pathlib
+import typing as t
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+
+class MissingMiddlewareError(Exception):
+    pass
+
+
+class ConnexionMiddleware:
+
+    default_middlewares = [
+    ]
+
+    def __init__(
+            self,
+            app: ASGIApp,
+            middlewares: t.Optional[t.List[t.Type[ASGIApp]]] = None
+    ):
+        """High level Connexion middleware that manages a list o middlewares wrapped around an
+        application.
+
+        :param app: App to wrap middleware around.
+        :param middlewares: List of middlewares to wrap around app. The list should be ordered
+                            from outer to inner middleware.
+        """
+        if middlewares is None:
+            middlewares = self.default_middlewares
+        self.app, self.apps = self._apply_middlewares(app, middlewares)
```

Comment:
If I understand correctly, the `apps` and `middlewares` will have the opposite direction, given that we iterate on the reversed middlewares?